### PR TITLE
MOB-3153 - Cancellation component

### PR DESCRIPTION
**Pull request by jsm20**
## JIRA
[MOB-3153](https://karhoo.atlassian.net/browse/MOB-3153)

## DESCRIPTION
Refactor and use the ContactOptionsView for cancellation. UI tweaks for Ride Details to be completed in the next PR.

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [x] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [x] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [x] Lint and unit tests pass locally (if appropriate)
- [x] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

File: uisdk/src/main/java/com/karhoo/uisdk/screen/rides/detail/RideDetailView.kt [hunk 1]
```diff
@@ -219,11 +209,11 @@ class RideDetailView @JvmOverloads constructor(
         if (cancellationDialog?.isShowing == true) {
             cancellationDialog?.dismiss()
         }
-        cancelRideButton.visibility = View.GONE
+        contactOptionsWidget.disableCancelButton()
     }
 
     override fun hideContactFleetButton() {
-        contactFleetButton.visibility = View.GONE
+        contactOptionsWidget.enableCancelButton()
```
Comment (mohadian) [inline uisdk/src/main/java/com/karhoo/uisdk/screen/rides/detail/RideDetailView.kt:216]: This logic does not seem to be working for TripDetails screen. When the Trip is completed, now we display 3 buttons and the Contact Fleet button is redundant. You need to wait for the trip to complete

Comment (jsm20) [reply uisdk/src/main/java/com/karhoo/uisdk/screen/rides/detail/RideDetailView.kt:216]: Do you mean Trip Details or Ride Details?

Comment (mohadian) [reply uisdk/src/main/java/com/karhoo/uisdk/screen/rides/detail/RideDetailView.kt:216]: Ride details

Comment (jsm20) [reply uisdk/src/main/java/com/karhoo/uisdk/screen/rides/detail/RideDetailView.kt:216]: Okay. Has been fixed in the UI PR. Will add a test there to confirm.

**Review by EdwardW2 — APPROVED** [review]

**Pull request merged**